### ISSUE TITLE
[UI] GraphQL Subscription for Test Results

### ISSUE THREAD

**Issue by leecalcote** [layer5io/meshery#2917]
#### Current Behavior
When a user is on the /performance page, runs a performance test and the results return, the count of performance tests does not visibly increment until the user refreshes the screen or navigates away and back again.

#### Desired Behavior
The performance test count should update dynamically. 

#### Implementation
This information should be supplied over a GraphQL subscription.

#### Acceptance Tests
1. Performance test count is updated w/o the user having to reload the page.

#### Mockups
<img width="488" alt="Screen Shot 2021-05-13 at 8 39 02 AM" src="https://user-images.githubusercontent.com/7570704/118147822-97bbb800-b3d5-11eb-9797-6320fa66680c.png">

---
**[Contributor Guide](https://docs.meshery.io/project/contributing)**

**Comment by leecalcote** [comment]
@srisatya1197 would you like to try this one?

**Comment by srisatya1197** [comment]
Sure @leecalcote Sir. But it may take some time. Do you have any problem with that , please let me know.

**Comment by leecalcote** [comment]
Thanks for setting expectations, @srisatya1197. If this works for you, let’s set a two week limit and bring in help or reassign at that time. Ultimately, the fix here is likely to include the need to enhance the existing graphql resolvers (in golang), too.

**Comment by srisatya1197** [comment]
I'm ready to do it but I want to make sure that everything goes well. Thanks for the motivation @leecalcote Sir.

**Comment by adithyaakrishna** [comment]
Hey @srisatya1197 , here's a clue https://github.com/meshery/meshery/blob/master/ui/components/Navigator.js . 
Happy Hacking 🚀

**Comment by srisatya1197** [comment]
@adithyaakrishna Thank you.

**Comment by srisatya1197** [comment]
I'll try to make a pr by sunday please @leecalcote Sir.

**Comment by srisatya1197** [comment]
@leecalcote Sir couldn't complete assigned work on time. Facing some issues with the setup and trying to fix them with the help  of adithya. I presume it may take time.

**Comment by leecalcote** [comment]
@srisatya1197 thanks much for this updated. @Gautam-Arora24, are you looking into this one?

**Comment by Gautam-Arora24** [comment]
Hi @leecalcote! Quoting Dhruv's reply from Slack.

>there might be a refactoring of the Meshery Server need to be done before the back-end GraphQL APIs are available. 

So I can start it as soon as the API's are available to me.

**Comment by sudo-NithishKarthik** [comment]
https://user-images.githubusercontent.com/75248557/128737153-615e2518-bd2a-4013-9479-65277a15edca.mov

**Comment by sudo-NithishKarthik** [comment]
Although it is not using graphql subscriptions, it is refetching the data when the modal is closed, hence the update in number of test results.

**Comment by sudo-NithishKarthik** [comment]
The only case i can think of that would require graphql subscriptions is when the test is ran with mesheryctl keeping ui open, in that case, we will get live data.
 Other than that, i don't think we would **_need_** graphql subscription for this **_now_**.

// @leecalcote

**Comment by SirZeck** [comment]
@SriSatyaT or @Gautam-Arora24 are you working on this issue? Seems inactive for quite long?

**Comment by Gautam-Arora24** [comment]
@SirZeck You can work on this one!

**Comment by sudo-NithishKarthik** [comment]
@bariqhibat , I am assuming it is already done. Am i right ?

**Comment by ThePiyushAggarwal** [comment]
@sudo-NithishKarthik 

I think @leecalcote  was talking about the Performance page

![image](https://user-images.githubusercontent.com/27003616/221379336-49d4d126-9495-44d5-9456-e78bd144d266.png)

I tried to recreate the issue. The test modal that is there is asking for creating a new test profile every time. And it was not able to complete any tests. I would just keep loading endlessly. The modal doesn't look like the same as in Profile page. The timer isn't there at all, just the loader. It won't stop even if the modal is closed. And there is no updation for the profile count in the page.

![image](https://user-images.githubusercontent.com/27003616/221379389-3e096c01-3fc0-4524-a33f-a4bff6afa16a.png)

**Comment by ishwar170695** [comment]
I looked into this, and I don't think this issue is still applicable.

- Meshery no longer supports GraphQL subscriptions. The schema explicitly notes that `Subscription `and the `graphql-ws` transport have been retired in favor of SSE/REST.
- Performance profiles now use RTK Query + REST (`performance-profile.ts`) instead of GraphQL subscriptions.
- The Performance page already fetches profile data through `useGetPerformanceProfilesQuery`, with updates handled via the current RTK Query flow.

Since the original issue is specifically about adding a GraphQL subscription for performance test counts, it no longer matches the current architecture. It seems appropriate to close it as obsolete unless there's still a missing real-time update requirement that should be tracked separately.
